subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyAdder.java

```diff
@@ -30,16 +30,14 @@
 /**
  * <p>A {@code DependencyAdder} is used to add dependencies to a specific configuration.</p>
  *
+ * These methods are not intended to be implemented by end users or plugin authors.
  * @since 7.6
  */
 @Incubating
 @NonExtensible
 public interface DependencyAdder {
-    /*
-    NOTE: If you add new dependency notations, you also need to consider how this affects these files:
-    - DependencyAdderExtensionModule.java
-    - DependencyAdderExtensions.kt
-    */
+    // See DependenciesExtensionModule for Groovy DSL
```

Comment:
🧐 Is this better suited for an `@apiNote`?  It would be more prominent on the javadoc for the type in any case, I think.

Reply:
This can't be made an apiNote or included in the Javadoc because these types aren't visible to core-api and are largely an implementation detail.  This is important for us to know, but users shouldn't care for the most part.